internal/resource_edge_firewall_edge_functions_instance.go

```diff
@@ -0,0 +1,437 @@
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"edge_firewall_id": schema.Int64Attribute{
+				Description: "The edge firewall identifier.",
+				Required:    true,
+			},
+			"schema_version": schema.Int64Attribute{
+				Computed: true,
+			},
+			"last_updated": schema.StringAttribute{
+				Description: "Timestamp of the last Terraform update of the resource.",
+				Computed:    true,
+			},
+			"results": schema.SingleNestedAttribute{
+				Required: true,
+				Attributes: map[string]schema.Attribute{
+					"id": schema.Int64Attribute{
+						Description: "The edge function instance identifier.",
+						Computed:    true,
+					},
+					"last_editor": schema.StringAttribute{
+						Description: "Last editor of the edge firewall edge functions instance.",
+						Computed:    true,
+					},
+					"last_modified": schema.StringAttribute{
+						Description: "Last modified timestamp of the edge firewall edge functions instance.",
+						Computed:    true,
+					},
+					"edge_function_id": schema.Int64Attribute{
+						Description: "The edge function identifier.",
+						Required:    true,
+					},
+					"name": schema.StringAttribute{
+						Description: "Name of the function.",
+						Required:    true,
+					},
+					"args": schema.StringAttribute{
+						Description: "JSON arguments of the function.",
+						Optional:    true,
+						Computed:    true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *edgeFirewallFunctionsInstanceResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	r.client = req.ProviderData.(*apiClient)
+}
+
+func (r *edgeFirewallFunctionsInstanceResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan edgeFirewallEdgeFunctionInstanceResourceModel
+	var edgeFirewallId types.Int64
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	diagsEdgeApplicationID := req.Config.GetAttribute(ctx, path.Root("edge_firewall_id"), &edgeFirewallId)
+	resp.Diagnostics.Append(diagsEdgeApplicationID...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var argsStr string
+	if plan.EdgeFunction.Args.IsUnknown() {
+		argsStr = "{}"
+	} else {
+		if plan.EdgeFunction.Args.ValueString() == "" || plan.EdgeFunction.Args.IsNull() {
+			resp.Diagnostics.AddError("Args",
+				"Is not null")
+			return
+		}
+		argsStr = plan.EdgeFunction.Args.ValueString()
+	}
+
+	planJsonArgs, err := utils.ConvertStringToInterface(argsStr)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			err.Error(),
+			"err",
+		)
+	}
+	if resp.Diagnostics.HasError() {
+		return
+	}
```

Comment:
What about returning earlier here?
```suggestion
	planJsonArgs, err := utils.ConvertStringToInterface(argsStr)
	if err != nil {
		resp.Diagnostics.AddError(
			err.Error(),
			"err",
		)
		return
	}
```